proxy/http/remap/NextHopSelectionStrategy.cc

```diff
@@ -108,7 +110,12 @@ NextHopSelectionStrategy::Init(ts::Yaml::Map &n)
         } else if (ring_mode_val == exhaust_rings) {
           ring_mode = NH_EXHAUST_RING;
         } else if (ring_mode_val == peering_rings) {
-          ring_mode = NH_PEERING_RING;
+          ring_mode            = NH_PEERING_RING;
+          YAML::Node self_node = failover_node["self"];
+          if (self_node) {
```

Comment:
@ywkaras Is this correct, `self` is part of the failover node`.  I'm not sure how this works in identifying the self host.  Maybe I'm missing something.  In testing, I'm looping when the self host is selected